### PR TITLE
f

### PR DESCRIPTION
**Pull request by XOctOpus1**

### CODE REVIEW

File: services/movie.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    if genres_ids is not None:
+        queryset = queryset.filter(genres__id__in=genres_ids)
+
+    if actors_ids is not None:
+        queryset = queryset.filter(actors__id__in=actors_ids)
+
+    return queryset
+
+
+def get_movie_by_id(movie_id: int) -> QuerySet[Movie]:
```
Comment (ArsenPidhoretskyi) [inline services/movie.py:21]: It returns one Movie, not queryset

File: services/movie.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    return Movie.objects.get(id=movie_id)
+
+
+def create_movie(
+        movie_title: str,
+        movie_description: str,
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
+) -> QuerySet[Movie]:
```
Comment (ArsenPidhoretskyi) [inline services/movie.py:30]: It returns Movie

File: services/movie_session.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+from db.models import MovieSession, Movie, CinemaHall
+from django.db.models.query import QuerySet
+from typing import Optional
+
+
+def create_movie_session(
+        movie_show_time: datetime.datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> QuerySet[MovieSession]:
```
Comment (ArsenPidhoretskyi) [inline services/movie_session.py:11]: it returns MovieSession

Comment (ArsenPidhoretskyi) [reply services/movie_session.py:11]: Please fix in all places

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: db/migrations/0002_cinemahall_movie_moviesession.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Generated by Django 4.0.2 on 2023-10-27 23:02
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('db', '0001_initial'),
```
Comment (vasylhnatiuk) [inline db/migrations/0002_cinemahall_movie_moviesession.py:10]: change ' to ", use Black

File: services/movie.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
+from db.models import Movie
+from django.db.models.query import QuerySet
+
+
+def get_movies(
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
+) -> QuerySet:
```
Comment (vasylhnatiuk) [inline services/movie.py:8]:
```suggestion
) -> QuerySet[Movie]:
```

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: db/migrations/0002_cinemahall_movie_moviesession.py (hunk 4 above)
Comment (XOctOpus1) [reply db/migrations/0002_cinemahall_movie_moviesession.py:10]: but there`s no need to change migrations. I thought that it`s better to let django make migrations

File: db/models.py [hunk 6]
```diff
@@ -14,3 +14,35 @@ class Actor(models.Model):
 
     def __str__(self) -> str:
         return f"{self.first_name} {self.last_name}"
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    actors = models.ManyToManyField(Actor, related_name="actors")
+    genres = models.ManyToManyField(Genre, related_name="genres")
```
Comment (vasylhnatiuk) [inline db/models.py:23]: wrong related name

File: db/models.py [hunk 7]
```diff
@@ -14,3 +14,35 @@ class Actor(models.Model):
 
     def __str__(self) -> str:
         return f"{self.first_name} {self.last_name}"
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    actors = models.ManyToManyField(Actor, related_name="actors")
```
Comment (vasylhnatiuk) [inline db/models.py:22]: wrong related name

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: db/models.py [hunk 8]
```diff
@@ -14,3 +14,35 @@ class Actor(models.Model):
 
     def __str__(self) -> str:
         return f"{self.first_name} {self.last_name}"
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    actors = models.ManyToManyField(Actor)
```
Comment (vasylhnatiuk) [inline db/models.py:22]:
```suggestion
    actors = models.ManyToManyField(Actor, related_name="movies")
```

File: db/models.py [hunk 9]
```diff
@@ -14,3 +14,35 @@ class Actor(models.Model):
...
+    genres = models.ManyToManyField(Genre)
```
Comment (vasylhnatiuk) [inline db/models.py:23]:
```suggestion
    genres = models.ManyToManyField(Genre, related_name="movies")
```

Comment (vasylhnatiuk) [reply db/models.py:23]: https://djangocentral.com/understanding-related-name-in-django-models/

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**